src/test/presentational-components/portfolio/__snapshots__/portfolios-filter-toolbar.test.js.snap

```diff
@@ -1,47 +1,79 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`<PortfoliosFilterToolbar /> should render correctly 1`] = `
-<Toolbar
-  className="pf-u-mt-md"
+<ContextProvider
+  value={
+    Object {
+      "componentMapper": Object {
+        "Button": [Function],
+        "FilterToolbarItem": [Function],
+        "Level": [Function],
+        "LevelItem": [Function],
+        "Link": [Function],
+        "Toolbar": [Function],
+        "ToolbarGroup": [Function],
+        "ToolbarItem": [Function],
+        "TopToolbar": [Function],
+        "TopToolbarTitle": [Function],
+      },
+      "render": [Function],
+    }
+  }
 >
-  <ToolbarGroup
-    className={null}
+  <Component
+    component="TopToolbar"
+    key="portfolios-top-toolbar"
   >
-    <ToolbarItem
-      className={null}
+    <Component
+      component="TopToolbarTitle"
+      key="portfolios-toolbar-title"
+      title="Portfolios"
+    />
+    <Component
+      className="pf-u-mt-md"
+      component="Toolbar"
+      key="main-portfolio-toolbar"
     >
-      <FilterToolbarItem
-        onFilterChange={[MockFunction]}
-        placeholder="Filter by name..."
-        searchValue=""
-      />
-    </ToolbarItem>
-  </ToolbarGroup>
-  <ToolbarGroup
-    className={null}
-  >
-    <ToolbarItem
-      className={null}
-    >
-      <Link
-        to="/portfolios/add-portfolio"
+      <Component
+        component="ToolbarGroup"
+        key="filter-group/single-toolbar-item-group"
+      >
+        <Component
+          component="ToolbarItem"
+          key="filter-group/single-toolbar-item"
+        >
+          <Component
+            component="FilterToolbarItem"
+            key="filter-input"
+            onFilterChange={[MockFunction]}
+            searchValue=""
+          />
+        </Component>
+      </Component>
+      <Component
+        component="ToolbarGroup"
+        key="portfolio-button-group/single-toolbar-item-group"
       >
-        <Button
-          aria-label="Create portfolio"
-          className=""
-          component="button"
-          isActive={false}
-          isBlock={false}
-          isDisabled={false}
-          isFocus={false}
-          isHover={false}
-          type="button"
-          variant="primary"
+        <Component
+          component="ToolbarItem"
```

Comment:
should this field be named something other 'component' - maybe 'type'?

Reply:
So this is just a generic component provider which holds the Context with component list that returns the actual React Component that should be rendered. You can actually reference React component directly instead of the string and that is what is happening in virtual dom. Yout don't see the string but React component from mapper. Unfortunately Jest is not smart enough to map the string to component into snapshots.

Reply:
ok, sounds good